core/aws-lsp-fqn/src/common/fqnExtractor.ts

```diff
@@ -0,0 +1,34 @@
+import { Extent, Java, Location, Python, Tsx, TypeScript } from '@aws/fully-qualified-names'
+import { FqnExtractorInput, FqnExtractorOutput } from './types'
+
+function extractNames(
+    languageId: FqnExtractorInput['languageId'],
+    fileText: string,
+    extent: Extent
+): Promise<FqnExtractorOutput> {
+    switch (languageId) {
+        case 'java':
+            return Java.findNamesWithInExtent(fileText, extent)
+        case 'javascriptreact':
+        case 'typescriptreact':
+            return Tsx.findNamesWithInExtent(fileText, extent)
+        case 'javascript':
+        case 'typescript':
+            return TypeScript.findNamesWithInExtent(fileText, extent)
```

Comment:
There is a known issue with the fqn package, where it hangs on various kinds of js/ts input. For that reason, we have currently disabled it for those filetypes in the Amazon Q extension: https://github.com/aws/aws-toolkit-vscode/pull/4954

Reply:
That's why we are running it in a separate worker thread with timeout.

Reply:
is that mentioned anywhere in the code, docs, etc?

Reply:
I am prepping a doc to explain the fqn usage in DEXP server, which just needs to be updated to reflect the newest changes, but I can also add to the readme here.